src/components/global/DialogForm.vue

```diff
@@ -0,0 +1,173 @@
+<script lang="ts">
+/**
+ * DialogForm Component
+ *
+ * The `DialogForm` component is used to render a dialog window that can be
+ * customized with various slots. Supports validation.
+ *
+ * It provides slots for the following content:
+ *
+ * - `title`: For the title of the dialog.
+ * - `content`: For the main content of the dialog.
+ * - `buttons`: For action buttons.
+ *
+ * The `content` and `image` slots are rendered side by side within
+ * a scrollable dialog window (on desktop screens).
+ *
+ * @props
+ * - `modelValue` (Boolean, required): Controls the visibility of the dialog.
+ * - `formRef` (Object, required): Template reference to the contained form.
+ *
+ * @events
+ * - `update:modelValue`: Emitted as a part of v-model structure
+ * - `change`: Emitted when the dialog is closed or opened
+ *
+ * @example
+ * <dialog-card>
+ *   <template v-slot:title>
+ *     <!-- Title content here -->
+ *   </template>
+ *   <template v-slot:content>
+ *     <!-- Main content here -->
+ *   </template>
+ *   <template v-slot:buttons>
+ *     <!-- Action buttons here -->
+ *   </template>
+ * </dialog-form>
+ *
+ * @see [Figma Design](https://www.figma.com/file/L8dVREySVXxh3X12TcFDdR/Do-pr%C3%A1ce-na-kole?type=design&node-id=6014%3A46223&mode=dev)
+ */
+
+// libraries
+import { defineComponent, computed } from 'vue';
+import { QForm } from 'quasar';
+
+export default defineComponent({
+  name: 'DialogForm',
+  props: {
+    formRef: {
+      type: Object as () => typeof QForm | null,
+      required: false,
+    },
+    modelValue: {
+      type: Boolean,
+      required: true,
+    },
+  },
+  emits: ['change', 'form-submit', 'update:modelValue'],
+  setup(props, { emit }) {
+    // determines if dialog window is open
+    const isOpen = computed({
+      get: (): boolean => props.modelValue,
+      set: (value: boolean): void => {
+        emit('update:modelValue', value);
+        emit('change');
+      },
+    });
+
+    // closes the dialog window
+    const close = (): void => {
+      if (props.formRef) {
+        props.formRef.reset();
+      }
+      isOpen.value = false;
+    };
+
+    /**
+     * Validates the form.
+     * If form is valid emits `form-submit` event and closes dialog.
+     */
+    const submit = async (): Promise<void> => {
+      if (props.formRef) {
+        const isFormValid: boolean = await props.formRef.validate();
+
+        if (isFormValid) {
+          emit('form-submit');
+          close();
+        } else {
+          props.formRef.$el.scrollIntoView({
+            behavior: 'smooth',
+          });
+        }
+      }
+    };
+
+    return {
+      isOpen,
+      close,
+      submit,
+    };
+  },
+});
+</script>
+
+<template>
+  <q-dialog square persistent v-model="isOpen" data-cy="dialog-form">
```

Comment:
Dialog component should looks like (base style) same across code, and we can reuse one base Dialog component for all use cases. See my comment below.

https://github.com/auto-mat/ride-to-work-by-bike-frontend/blob/270efb98d2830ab25af92092282b63e86280fbdc/src/components/global/DialogCard.vue#L77

Reply:
Reworked into a `DialogDefault` component.